kartothek/io/dask/dataframe.py

```diff
@@ -606,3 +604,67 @@ def hash_dataset(
             .apply(_unpack_hash, unpack_meta=ddf._meta, subset=subset, meta="uint64")
             .astype("uint64")
         )
+
+
+def calculate_repartition_ratio(
+    dataset_factory: DatasetFactory,
+    predicates: Optional[PredicatesType] = None,
+    *,
+    target_nrows_per_partition: int,
+    dataset_metadata_frac: float,
+) -> dd.core.Scalar:
+    """
+    Calculate a `repartition_ratio` according to the dataset input and `target_nrows_per_partition` (the desired
+    number of rows of each dataframe partition).
+
+
+    Parameters
+    ----------
+    dataset_factory
+        dataset factory
+    predicates
+        Optional , but predicates to be applied during calculatin repartition ratio
+    target_nrows_per_partition
+        target num of rows per partition
+    dataset_metadata_frac
+        used during collection of dataset metadata
+
+    Returns
+    -------
+    repartition_ratio
+        calculated repartition_ratio
+    """
+    dataset_metadata = collect_dataset_metadata(
+        factory=dataset_factory, predicates=predicates, frac=dataset_metadata_frac,
+    )
+    expected_nrows_per_partition = dataset_metadata.groupby("partition_label")[
+        "number_rows_per_row_group"
+    ].sum()
+
+    mean_nrows_per_partition = expected_nrows_per_partition.mean()
+    repartition_ratio = target_nrows_per_partition / mean_nrows_per_partition
+    return repartition_ratio
+
+
+def repartition_ddf(
+    ddf: dd.DataFrame, repartition_ratio: SupportsFloat
+) -> dd.DataFrame:
+    """
+    Repartition a dask DataFrame according, dividing the current number of partitions by `repartition_ratio`.
```

Comment:
What exactly is a "repartition ratio"? Neither this nor the other function explain this term.

Reply:
I should have read the PR comments before, sorry. I see you already more or less agreed that this term is somewhat fragile / non-user-friendly.

Reply:
The logic is updated completely. Please review new logic and give suggestions.